nixos/lib/make-ext4-fs.nix

```diff
@@ -78,6 +78,15 @@ pkgs.stdenv.mkDerivation {
       # get rid of the unnecessary slack here--but see
       # https://github.com/NixOS/nixpkgs/issues/125121 for caveats.
 
+      # shrink to fit
+      resize2fs -M $img
+
+      # Add 16 MebiByte to the current_size
+      new_size=$(dumpe2fs -h $img | awk -F: \
+        '/Block count/{count=$2} /Block size/{size=$2} END{print (count*size+16*2**20)/size}')
```

Comment:
Why?

Reply:
My understanding is, that the shrinked image might not have enough space for the first boot to succeed, as some files are created during early boot __before__ expanding the partition. This ensures that the partition is resized to be 16 MebiBytes bigger than needed, thus mitigating the problem.